public/app/features/browse-dashboards/state/actions.ts

```diff
@@ -1,10 +1,89 @@
-import { createAsyncThunk } from '@reduxjs/toolkit';
-
+import { getBackendSrv } from '@grafana/runtime';
+import { DeleteDashboardResponse } from 'app/features/manage-dashboards/types';
+import { GENERAL_FOLDER_UID } from 'app/features/search/constants';
 import { getFolderChildren } from 'app/features/search/service/folders';
+import { createAsyncThunk, DashboardDTO } from 'app/types';
+
+import { rootItemsSelector, childrenByParentUIDSelector } from './hooks';
+import { findItem } from './utils';
 
 export const fetchChildren = createAsyncThunk(
   'browseDashboards/fetchChildren',
   async (parentUID: string | undefined) => {
-    return await getFolderChildren(parentUID, undefined, true);
+    // Need to handle the case where the parentUID is the root
+    const uid = parentUID === GENERAL_FOLDER_UID ? undefined : parentUID;
+    return await getFolderChildren(uid, undefined, true);
```

Comment:
If I'm reading this correctly - are we getting a parentUID of `general` on some items? That doesn't seem right...

Reply:
yeah seems i am locally at least 😬 agreed this bit feels very ugly